internal/pkg/deploy/cloudformation/stack/backend_svc.go

```diff
@@ -208,6 +209,23 @@ func (s *BackendService) Template() (string, error) {
 	return string(overriddenTpl), nil
 }
 
+func (s *BackendService) getPrimaryContainerPortMappings() []*template.PortMapping {
+	containerPort := aws.String(s.containerPort())
+	var portMappings []*template.PortMapping
+	if aws.StringValue(containerPort) != NoExposedContainerPort {
+		portMappings = append(portMappings, ConvertPortMapping(containerPort, aws.String("tcp"), s.name))
+	}
+	if s.isTargetPortDifferentThanPrimaryContainerPort(containerPort) {
```

Comment:
hopefully this method will get removed and most of the remaining logic will be simpler after the change proposed above 👍